### DISCUSSION TITLE
AngularJS e SOAP

### DISCUSSION THREAD

**Discussion by hugocarreira** [frontendbr/forum#469] [Discussão]
Pessoal, estou com um desafio no trabalho que é integrar o AngularJS (sim, 1.x) e um webservice SOAP.
Sei que o ideial seria REST porém o SOAP são ordens superiores, então não tem muito o que reclamar.

Alguém já passou por algo parecido? Se sim, como trabalhou? Alguém já viu alguma coisa?

Sei que tem um módulo AngularSOAP para isso, porém está muito imaturo e cheio de bugs..

Estou realmente meio perdido..

**Comment by ojonathanleal-zz** [comment]
@hugocarreira Eu já trabalhei em um projeto com essa necessidade. O que fizemos foi colocar uma camada de controllers no back-end e consumi-lá com o Angular utilizando REST. Essa camada de controller que ficaria encarregada de acessar o serviço SOAP através de algum cliente e transformar para JSON para retornar para o Angular. Assim você pode escolher alguma lib da linguagem que você escolher para colocar os controllers para consumir serviços SOAP. Sacas?

**Comment by robsonbittencourt** [comment]
Trabalho em um projeto dessa forma também. Existe uma camada REST que
consome os SOAP.

On Wed, Feb 22, 2017 at 5:38 PM Jonathan Leal <notifications@github.com>
wrote:

> @hugocarreira <https://github.com/hugocarreira> Eu já trabalhei em um
> projeto com essa necessidade. O que fizemos foi colocar uma camada de
> controller no back-end e consumi-lá com o angular. Essa camada de
> controller que ficaria encarregada de acessar o serviço SOAP através de
> algum cliente. Assim você pode escolher alguma lib da linguagem que você
> escolher para colocar os controllers. Sacas?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/frontendbr/forum/issues/469#issuecomment-281795693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABMNWjlMBVMrKlDycNpomhMRPI0hHbAyks5rfJy3gaJpZM4MIogM>
> .
>
-- 
Robson Bittencourt
Desenvolvedor de Software
